docs/openapi-user.yaml

```diff
@@ -1,170 +1,419 @@
+    ProductId:
+      type: string
+      format: ulid
+    Timestamp:
+      type: string
+      format: UTCISODateFromString
+      description: A date-time field in ISO-8601 format and UTC timezone.
+      x-import: "@pagopa/ts-commons/lib/dates"
+      example: "2018-10-13T00:00:00.000Z"
+    QrCodeUrl:
+      type: string
+      format: url
+      minLength: 1
 
-components:
-  schemas:
-    $ref: "./openapi-definitions.yaml"
+    SignatureRequestStatus:
+      type: string
+      x-extensible-enum:
+        - DRAFT
+        - WAIT_FOR_ISSUER
+        - READY
+        - WAIT_FOR_SIGNATURE
+        - SIGNED
+      description: >
+        Signature request status:
+         * `DRAFT` - The signature request has been created but not all documents have been uploaded yet.
+         * `WAIT_FOR_ISSUER` - All documents have been uploaded but the ISSUER has not yet marked the signature request as READY.
+         * `READY` - The signature request is ready for processing. In this phase the documents will be analyzed and prepared for sending to the USER.
+         * `WAIT_FOR_SIGNATURE` - The signature request has been sent to the user and is awaiting signature
+         * `SIGNED` - The signature request was successfully signed.
+
+    SignatureRequestDetailView:
+      type: object
+      properties:
+        id:
+          type: string
+          format: ulid
+        status:
+          $ref: "#/components/schemas/SignatureRequestStatus"
+        productId:
+          $ref: "#/components/schemas/ProductId"
+        expiresAt:
+          $ref: "#/components/schemas/Timestamp"
+        qrCodeUrl:
+          $ref: "#/components/schemas/QrCodeUrl"
+        documents:
+          type: array
+          items:
+            $ref: "#/components/schemas/Document"
+      required:
+        - id
+        - productId
+        - qrCodeUrl
+        - documents
+    SignatureRequestId:
+      type: string
+      format: ulid
+      minLength: 1
+
+    QtspClauses:
+      type: object
+      properties:
+        clauses:
+          type: array
+          items:
+            $ref: "#/components/schemas/QtspClause"
+          minItems: 1
+          uniqueItems: true
+        mrcDocumentUrl:
+          type: string
+          format: uri
+        privacyDocumentUrl:
+          type: string
+          format: uri
+        tosDocumentUrl:
+          type: string
+          format: uri
+      required:
+        - text
+        - mrcUrl
+
+    QtspClause:
+      type: object
+      properties:
+        text:
+          type: string
+      required:
+        - text
+
+    PublicKey:
+      type: string
+      format: base64url
+      description: PEM public key encoded in base64
+
+    SamlAssertion:
+      type: string
+      format: base64url
+      description: SPID SAML assertion encoded in base64
+
+    ClausesSignature:
+      type: object
+      properties:
+        signatureFieldId:
+          type: string
+        accepted:
+          type: boolean
```

Comment:
The clauses must not also have a required field boolean? Should the document clauses in the UI not be mandatory or optional?

Reply:
Yes, to show them you have to retrieve them from the `signature-request`. In this case it is the POST of the accepted clauses where it is not necessary to know if they are mandatory or not.

Reply:
Great! Thanks!